metric-example/query-by-example.md

```diff
@@ -0,0 +1,516 @@
+
+# Scenario 13: Combine Series and Point Queries with Folding
+
+>**Task**
+>For a report, we want to query both per-month CPU usage, but also get an overall average.
+
+In some situations, it is useful to transmute a series result into a point result in any position of the transformer chain. One such example is when we want mixed series and point results.
+
+Say, we want to access `builtin:host.cpu.usage` over the complete last year from January to just before January this year (`now-y/y` to `now/y`). We want a resolution of `1M`, but we are also interested in the average value over the whole year. This can be done in a single bulk query, once using `:fold` and once omitting it:
+```
+GET {base}/metrics/query?metricId=builtin:host.cpu.usage:fold,builtin:host.cpu.usage&from=now-y/y&to=now/y&resolution=1M
+```
+# Scenario 14: Maximum of Average CPU Usage Values Over Time
+
+>**Task**
+>For a chart, we want to draw a line that exactly intersects the peak of the graph.
+
+When can apply what we learned above and query for a series of the average over time and simultaneously query for the maximum over time:
+```
+builtin:host.cpu.usage:avg  
+builtin:host.cpu.usage:fold:max
+```
+The second line merges the statistical data over all time slots, and then extracts the maximum value from the statistical data. This works, but the folded value will be actual maximum sample. If we draw the series and the maximum together into a chart, the folded value may exceed the highest average value, since each value in the chart is actually an aggregate of multiple samples (the average of samples in our case). What if we want to query for the maximum of average values in the chart, not the actual maximum sample?
+
+For that use case, we want to build a list of averages rather than a list of original samples, and extract the maximum from these averages.
+
+We can take advantage of the fact that `:splitBy` collects pre-aggregated input values (e.g. averages) into a list if the next aggregation after the merge requires it. If we pass all dimensions of the metric to `:splitBy,` no dimensions will be removed and hence no actual merging will happen, instead we have a list with a single average for each time slots. Before the final aggregation, we need to insert `:fold` so that the lists are merged into one over time. From this merged list, we finally extract the maximum with `:max`. Taking all of this together we end up with this selector:
+```
+builtin:host.cpu.usage:avg:splitBy(dt.entity.host):fold:max
+```
+The result is exactly what we wanted: the maximum of average values over time - We can draw the maximum into the chart and see that it will exactly touch the peak of the graph.

```

Comment:
```suggestion
The result is exactly what we wanted: the maximum average values over time. We can chart the maximum and see that it will exactly touch the peak of the graph.

```